core/src/wallet/tezos/explorers/BakingBadTezosLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,471 @@
+            std::string params = fmt::format("?limit={}", limit);
+            if (localOffset > 0) {
+                params += fmt::format("&lastId={}", localOffset);
+            }
+            using EitherTransactionsBulk = Either<Exception, std::shared_ptr<TransactionsBulk>>;
+            return _http->GET(fmt::format("v1/accounts/{}/operations{}", addresses[0], params))
+                .template json<TezosLikeBlockchainExplorer::TransactionsBulk, Exception>()
+                .template mapPtr<TransactionsBulk>(getExplorerContext(),
+                                                   [](const EitherTransactionsBulk &result) {
+                                                       if (result.isLeft()) {
+                                                           throw result.getLeft();
+                                                       }
+                                                       result.getRight()->hasNext = result.getRight()->transactions.size() == limit;
+                                                       return result.getRight();
+                                                   })
+                .flatMapPtr<TransactionsBulk>(getExplorerContext(), AddPublicKeyToRevealTx(addresses));
+        }
+
+        FuturePtr<Block> BakingBadTezosLikeBlockchainExplorer::getCurrentBlock() const {
+            return _http->GET("v1/blocks?sort.desc=id&limit=1")
+                .template json<Block, Exception, TzKTParser>()
+                .template mapPtr<Block>(getExplorerContext(),
+                                        [](const Either<Exception, std::shared_ptr<Block>> &result) {
+                                            if (result.isLeft()) {
+                                                throw result.getLeft();
+                                            }
+                                            return result.getRight();
+                                        });
+        }
+
+        FuturePtr<TezosLikeBlockchainExplorerTransaction>
+        BakingBadTezosLikeBlockchainExplorer::getTransactionByHash(const String &transactionHash) const {
+            return getLedgerApiTransactionByHash(transactionHash);
+        }
+
+        Future<int64_t> BakingBadTezosLikeBlockchainExplorer::getTimestamp() const {
+            return getLedgerApiTimestamp();
+        }
+
+        std::shared_ptr<api::ExecutionContext> BakingBadTezosLikeBlockchainExplorer::getExplorerContext() const {
+            return _executionContext;
+        }
+
+        api::TezosLikeNetworkParameters BakingBadTezosLikeBlockchainExplorer::getNetworkParameters() const {
+            return _parameters;
+        }
+
+        std::string BakingBadTezosLikeBlockchainExplorer::getExplorerVersion() const {
+            return "";
+        }
+
+        Future<std::shared_ptr<BigInt>>
+        BakingBadTezosLikeBlockchainExplorer::getEstimatedGasLimit(const std::string & /*address*/) {
+            return FuturePtr<BigInt>::successful(
```

Comment:
I know that you copied this logic from previous code but, wow, this is scary. I can see that this code is actually used from [here](https://github.com/LedgerHQ/ledger-wallet-daemon/blob/bb296d92d82d225d3adf3a3b2512fa57dc91f3d5/src/main/scala/co/ledger/wallet/daemon/models/Account.scala#L517) and returning hardcoded value is scary. Apparently this is used only when crafting "non-transaction" transactions ^^" So it should not happen with the current use cases...

Reply:
I agree it's scary, but what change do you want here ? :/

Reply:
Nothing, just to keep that in mind.. :)